cache/async_queue_test.go

```diff
@@ -0,0 +1,51 @@
+package cache
+
+import (
+	"testing"
+	"time"
+
+	"go.uber.org/atomic"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestAsyncQueue_Run(t *testing.T) {
+	q := newAsyncQueue(10, 10)
+	defer q.stop()
+
+	var i atomic.Int32
+	_ = q.run(func() { i.Add(1) })
+	_ = q.run(func() { i.Add(-1) })
+	_ = q.run(func() { i.Add(1) })
+	_ = q.run(func() { i.Add(-1) })
+	_ = q.run(func() { i.Add(1) })
+
+	// Wait for all operations to finish.
+	time.Sleep(100 * time.Millisecond)
```

Comment:
Better to call `q.stop()` here. No need to `time.Sleep()` and we also test `stop()` waits until all workers have done.

Reply:
I think your suggestion exposes a potential problem with the current implementation: not everything that has been queued will be executed if `.stop()` is called immediately due to the `select` statement in the worker loop.

Reply:
Right, it's a valid concern.